mijia/src/session.rs

```diff
@@ -0,0 +1,146 @@
+use anyhow::Context;
+use bluez_generated::bluetooth_event::BluetoothEvent;
+use bluez_generated::generated::{OrgBluezAdapter1, OrgBluezDevice1};
+use core::fmt::Debug;
+use core::future::Future;
+use core::time::Duration;
+use dbus::{
+    nonblock::{MsgMatch, SyncConnection},
+    Message,
+};
+use futures::{FutureExt, Stream, StreamExt};
+use std::sync::Arc;
+
+// TODO before publishing to crates.io: annotate this enum as non-exhaustive.
+#[derive(Clone)]
+pub enum MijiaEvent {
+    // FIXME: stop using object_path as primary key. Can we think of something better?
+    Readings {
+        object_path: String,
+        readings: crate::Readings,
+    },
+    Disconnected {
+        object_path: String,
+    },
+}
+
+impl MijiaEvent {
+    fn from(conn_msg: Message) -> Option<Self> {
+        match BluetoothEvent::from(conn_msg) {
+            Some(BluetoothEvent::Value { object_path, value }) => {
+                // TODO: Make this less hacky.
+                let object_path = object_path
+                    .strip_suffix(crate::SERVICE_CHARACTERISTIC_PATH)?
+                    .to_owned();
+                let readings = crate::decode_value(&value)?;
```

Comment:
Why not just use these symbols, rather than having crate:: all over the place?

Reply:
mostly because it's easier when doing experimental refactoring, because you don't need to keep your imports up-to-date with your code manually. I will stop being lazy and fix them.